internal/controller/appserver/deployment.go

```diff
@@ -494,13 +494,51 @@ func updateOLSDeployment(r reconciler.Reconciler, ctx context.Context, existingD
 		existingDeployment.Annotations[utils.OLSConfigHashKey] != r.GetStateCache()[utils.OLSConfigHashStateCacheKey] ||
 		existingDeployment.Annotations[utils.OLSAppTLSHashKey] != r.GetStateCache()[utils.OLSAppTLSHashStateCacheKey] ||
 		existingDeployment.Annotations[utils.LLMProviderHashKey] != r.GetStateCache()[utils.LLMProviderHashStateCacheKey] ||
-		existingDeployment.Annotations[utils.PostgresSecretHashKey] != r.GetStateCache()[utils.PostgresSecretHashStateCacheKey] {
+		existingDeployment.Annotations[utils.PostgresSecretHashKey] != r.GetStateCache()[utils.PostgresSecretHashStateCacheKey] ||
+		existingDeployment.Annotations[utils.PostgresCAHashKey] != r.GetStateCache()[utils.PostgresCAHashStateCacheKey] {
+
+		// Check if PostgreSQL-related hashes changed - if so, verify PostgreSQL deployment is ready before restarting app-server
+		existingCAHash := existingDeployment.Annotations[utils.PostgresCAHashKey]
+		newCAHash := r.GetStateCache()[utils.PostgresCAHashStateCacheKey]
+		existingSecretHash := existingDeployment.Annotations[utils.PostgresSecretHashKey]
+		newSecretHash := r.GetStateCache()[utils.PostgresSecretHashStateCacheKey]
+
+		postgresCAHashChanged := existingCAHash != newCAHash
+		postgresSecretHashChanged := existingSecretHash != newSecretHash
+		shouldWaitForPostgres := (postgresCAHashChanged || postgresSecretHashChanged) &&
+			!(existingCAHash == "" && existingSecretHash == "")
+
+		if shouldWaitForPostgres {
+			postgresDeployment := &appsv1.Deployment{}
+			err := r.Get(ctx, client.ObjectKey{Name: utils.PostgresDeploymentName, Namespace: r.GetNamespace()}, postgresDeployment)
+			if err == nil {
+				// PostgreSQL deployment exists, check if it's ready
+				if postgresDeployment.Status.ReadyReplicas != *postgresDeployment.Spec.Replicas {
+					// PostgreSQL is not ready yet, skip app-server update to avoid readiness failures
+					changeType := ""
+					if postgresCAHashChanged && postgresSecretHashChanged {
+						changeType = "PostgreSQL CA and secret"
+					} else if postgresCAHashChanged {
+						changeType = "PostgreSQL CA"
+					} else {
+						changeType = "PostgreSQL secret"
+					}
```

Comment:
The computation of what changes seems to be over kill - just for the log message - Could you please simplify ?